File: problems/algorithms/0509/C/solution_iterative_no_forgetting.c
int calculated[31] = {0};

int fib(int n)
{
	if (n < 2)
		return n;
	calculated[1] = 1;
	for (int i = 2; i <= n; i++)
	{
		calculated[i] = calculated[i - 1] + calculated[i - 2];
	}
	return calculated[n];
}

File: problems/algorithms/0509/Rust/solution_iterative_no_forgetting.rs
impl Solution {
    pub fn fib(n: i32) -> i32 {
        if n < 2 {
            return n;
        }
        let c = n as usize;
        let mut calculated = vec![0; c + 1];
        calculated[1] = 1;
        for i in 2..=c {
            calculated[i] = calculated[i - 1] + calculated[i - 2];
        }
        calculated[c]
    }
}